commit a936516107579df13362f6d2d6f91f26a447d6dd
Author: Martin Gräßlin <mgraesslin@kde.org>
Date:   Mon Apr 3 06:53:05 2017 +0200

    [kcmkwin/compositing] Do not write GLPlatformInterface config
    
    Summary:
    The gui element to select the platform interface is dropped. Thus we
    should not store the config for it. This is rather important as changing
    compositor settings on Wayland would result in egl being written into the
    config which potentially breaks an X11 session.
    
    BUG: 378114
    
    Reviewers: #kwin, #plasma
    
    Subscribers: plasma-devel, kwin
    
    Tags: #kwin
    
    Differential Revision: https://phabricator.kde.org/D5285

diff --git a/kcmkwin/kwincompositing/compositing.cpp b/kcmkwin/kwincompositing/compositing.cpp
index c1cddb89d..102dfd18c 100644
--- a/kcmkwin/kwincompositing/compositing.cpp
+++ b/kcmkwin/kwincompositing/compositing.cpp
@@ -309,10 +309,6 @@ void Compositing::save()
     }
     kwinConfig.writeEntry("Backend", backend);
     kwinConfig.writeEntry("GLCore", glCore);
-    const QModelIndex glIndex = m_openGLPlatformInterfaceModel->index(m_openGLPlatformInterface);
-    if (glIndex.isValid()) {
-        kwinConfig.writeEntry("GLPlatformInterface", glIndex.data(Qt::UserRole).toString());
-    }
     if (!compositingRequired()) {
         kwinConfig.writeEntry("WindowsBlockCompositing", windowsBlockCompositing());
     }
